app/models/liabilitycalculation/AllowancesAndDeductions.scala

```diff
@@ -0,0 +1,69 @@
+ * limitations under the License.
+ */
+
+package models.liabilitycalculation
+
+import play.api.libs.json._
+import play.api.libs.functional.syntax._
+
+case class AllowancesAndDeductions(
+                                    personalAllowance: Option[Int] = None,
+                                    marriageAllowanceTransferOut: Option[MarriageAllowanceTransferOut] = None,
+                                    reducedPersonalAllowance: Option[Int] = None,
+                                    giftOfInvestmentsAndPropertyToCharity: Option[Int] = None,
+                                    lossesAppliedToGeneralIncome: Option[Int] = None,
+                                    qualifyingLoanInterestFromInvestments: Option[BigDecimal] = None,
+                                    postCessationTradeReceipts: Option[BigDecimal] = None,
+                                    paymentsToTradeUnionsForDeathBenefits: Option[BigDecimal] = None,
+                                    grossAnnuityPayments: Option[BigDecimal] = None,
+                                    pensionContributions: Option[BigDecimal] = None
+                                  )
+
+object AllowancesAndDeductions {
+  implicit val writes: OWrites[AllowancesAndDeductions] = (
+    (JsPath \ "personalAllowance").writeNullable[Int] and
+      (JsPath \ "marriageAllowanceTransferOut").writeNullable[MarriageAllowanceTransferOut] and
+      (JsPath \ "reducedPersonalAllowance").writeNullable[Int] and
+      (JsPath \ "giftOfInvestmentsAndPropertyToCharity").writeNullable[Int] and
+      (JsPath \ "lossesAppliedToGeneralIncome").writeNullable[Int] and
+      (JsPath \ "qualifyingLoanInterestFromInvestments").writeNullable[BigDecimal] and
+      (JsPath \ "post-cessationTradeReceipts").writeNullable[BigDecimal] and
+      (JsPath \ "paymentsToTradeUnionsForDeathBenefits").writeNullable[BigDecimal] and
+      (JsPath \ "grossAnnuityPayments").writeNullable[BigDecimal] and
+      (JsPath \ "pensionContributions").writeNullable[BigDecimal]
+    ) (unlift(AllowancesAndDeductions.unapply))
+
+  implicit val reads: Reads[AllowancesAndDeductions] = (
+    (JsPath \ "personalAllowance").readNullable[Int] and
+      (JsPath \ "marriageAllowanceTransferOut").readNullable[MarriageAllowanceTransferOut] and
+      (JsPath \ "reducedPersonalAllowance").readNullable[Int] and
+      (JsPath \ "giftOfInvestmentsAndPropertyToCharity").readNullable[Int] and
+      (JsPath \ "lossesAppliedToGeneralIncome").readNullable[Int] and
+      (JsPath \ "qualifyingLoanInterestFromInvestments").readNullable[BigDecimal] and
+      (JsPath \ "post-cessationTradeReceipts").readNullable[BigDecimal] and
+      (JsPath \ "paymentsToTradeUnionsForDeathBenefits").readNullable[BigDecimal] and
+      (JsPath \ "grossAnnuityPayments").readNullable[BigDecimal] and
+      (JsPath \ "pensionContributions").readNullable[BigDecimal]
+    ) (AllowancesAndDeductions.apply _)
+}
+
+case class MarriageAllowanceTransferOut(
```

Comment:
I would put this case class and the companion object in a MarriageAllowanceTransferOut.scala file.

Reply:
Same as below: In my view there is not sufficient complexity to warrant a separate file for this one (at the cost of maintenance overhead and lower readability)